src/agni/client/receiver/MessageParser.java

```diff
@@ -1,5 +1,24 @@
 package agni.client.receiver;
 
+import agni.client.view.*;
+
 public abstract class MessageParser {
+    private ReceiverListener[] listeners;
+
+    public MessageParser() {
+        super();
+        // TODO Auto-generated constructor stub
+    }
+
+    public void register(AgniClientView listener) {
+
+    }
+
+    public void deregister(AgniClientView listener) {
+
+    }
+
+    public void receiveMessage(String message) {
```

Comment:
This method should have no implementation, this is what will be overridden by the Receiver classes.